web/packages/teleterm/src/ui/DocumentTerminal/useDocumentTerminal.ts

```diff
@@ -188,18 +312,23 @@ async function createPtyProcess(
 }
 
 function createCmd(
-  doc: Doc,
+  doc: types.DocumentTerminal,
   proxyHost: string,
   clusterName: string
 ): PtyCommand {
   if (doc.kind === 'doc.terminal_tsh_node') {
+    if (!('serverId' in doc)) {
+      return;
```

Comment:
In the next PR I changed this to throwing an error since the code shouldn't get to this point if doc is still a `DocumentTshNodeWithLoginHost`.